domenetjenester/tilgangskontroll/src/main/java/no/nav/ung/sak/tilgangskontroll/tilganger/TilgangTilOperasjonTjeneste.java

```diff
@@ -0,0 +1,101 @@
+package no.nav.ung.sak.tilgangskontroll.tilganger;
+
+import jakarta.enterprise.context.Dependent;
+import no.nav.k9.felles.sikkerhet.abac.BeskyttetRessursActionAttributt;
+import no.nav.ung.kodeverk.behandling.aksjonspunkt.AksjonspunktType;
+import no.nav.ung.sak.tilgangskontroll.TilgangsbeslutningInput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+@Dependent
+public class TilgangTilOperasjonTjeneste {
+
+    private final static Logger logger = LoggerFactory.getLogger(TilgangTilOperasjonTjeneste.class);
+
+    public Set<IkkeTilgangÅrsak> sjekkTilgangTilOperasjon(TilgangerBruker tilganger, TilgangsbeslutningInput.Operasjon operasjon, TilgangsbeslutningInput.Saksinformasjon saksinformasjon) {
+        try {
+            return internSjekkTilgangTilOperasjon(tilganger, operasjon, saksinformasjon);
+        } catch (Exception e) {
+            logger.warn("Fikk teknisk feil i tilgangssjekk, gir derfor ikke tilgang til operasjonen", e);
+            return Set.of(IkkeTilgangÅrsak.TEKNISK_FEIL);
+        }
+    }
+
+    Set<IkkeTilgangÅrsak> internSjekkTilgangTilOperasjon(TilgangerBruker tilganger, TilgangsbeslutningInput.Operasjon operasjon, TilgangsbeslutningInput.Saksinformasjon saksinformasjon) {
+        return switch (operasjon.getResource()) {
+            case FAGSAK -> sjekkTilgangTilFagsak(tilganger, operasjon.getAction(), saksinformasjon);
+            case VENTEFRIST -> sjekkTilgangTilVentefrist(tilganger);
+            case APPLIKASJON -> sjekkTilgangTilApplikasjon(tilganger);
+            case DRIFT -> sjekkTilgangTilDrift(tilganger);
+        };
+    }
+
+    private Set<IkkeTilgangÅrsak> sjekkTilgangTilFagsak(TilgangerBruker tilganger, BeskyttetRessursActionAttributt action, TilgangsbeslutningInput.Saksinformasjon saksinformasjon) {
+        return switch (action) {
+            case READ -> sjekkTilgangTilLesFagsak(tilganger);
+            case UPDATE -> sjekkTilgangTilOppdaterFagsak(tilganger, saksinformasjon);
+            case CREATE -> sjekkTilgangTilOpprettFagsak(tilganger);
+            case DELETE -> throw new IllegalArgumentException("Action DELETE er ikke i bruk");
+            case DUMMY -> throw new IllegalArgumentException("Action DUMMY er ikke i bruk");
+        };
+    }
+
+    private static Set<IkkeTilgangÅrsak> sjekkTilgangTilLesFagsak(TilgangerBruker tilganger) {
+        return tilganger.kanVeilede() || tilganger.kanSaksbehandle() ? Set.of() : Set.of(IkkeTilgangÅrsak.HAR_IKKE_TILGANG_Å_LESE_FAGSAK);
```

Comment:
Burde ikke dette settet inneholde grunnen til at man ikke kan lese fagsak for bedre feilmelding?
For eksempel hvis bruker ikke kan veilede, så burde årsaken inneholde noe ala. KAN_IKKE_VEILEDE.